### PR TITLE
PA-13820 Support regex for authVerificationURL

### PR DESCRIPTION
**Pull request by SOOS-MMalony**
### Changes:

Support regex for authVerificationURL

**Ticket:** https://soos.atlassian.net/browse/PA-13820

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "soos-dast",
-  "version": "2.0.31",
+  "version": "2.0.32",
```
Comment (SOOS-MMalony) [inline package.json:3]: Assuming I should bump this since I updated index.ts for arg description.

File: src/zap_hooks/helpers/auth.py [hunk 2]
```diff
@@ -114,9 +114,9 @@ def validate_authentication_url(driver, url):
     log(f"Validating authentication url: {url}")
     url_found = False
     for request in driver.requests:
-        if request.response and url in request.url:
+        if request.response and (url in request.url or search(url, request.url) is not None):
```
Comment (SOOS-MMalony) [inline src/zap_hooks/helpers/auth.py:117]: This supports regular URL (current functionality) or a regex expression.

Comment (SOOS-MMalony) [reply src/zap_hooks/helpers/auth.py:117]: Will validate all requests that match regex, not just the first. It's a quick change if we'd prefer to only validate the first match.

**Review by SOOS-GSteen — DISMISSED** [review]

File: src/zap_hooks/helpers/auth.py [hunk 3]
```diff
@@ -114,14 +114,15 @@ def validate_authentication_url(driver, url):
...
             url_found = True
-            log(f"Checking response status code {request.response}")
+            log(f"Checking response status code {request.response} for {request.url}")
             if request.response.status_code not in [200, 302]:
                 log(f"Status code is not 200/302 for {request.url}, it is {request.response.status_code}")
                 sys.exit(1)
             else: 
                 log(f"Status code is {request.response.status_code} for {request.url}, authentication was successful")
+                break
```
Comment (SOOS-MMalony) [inline src/zap_hooks/helpers/auth.py:125]: Based on feedback from @SOOS-FAngelino, only validating first matching url.

**Review by SOOS-GSteen — APPROVED** [review]

**Pull request merged**